server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/IdentityRepositoryTest.scala

```diff
@@ -135,11 +134,47 @@ class IdentityRepositoryTest {
   @Test
   def updateShouldSuccessWhenCustomExists(): Unit = {
     when(identityFactory.listIdentities(BOB)).thenReturn(List())
-    SMono.fromPublisher(customIdentityDAO.save(BOB, CREATION_REQUEST)).block()
-    val identity: Identity = SFlux(customIdentityDAO.list(BOB)).blockFirst().get
+    val customIdentity: Identity = SMono.fromPublisher(customIdentityDAO.save(BOB, CREATION_REQUEST)).block()
 
-    assertThatCode(() => SMono.fromPublisher(testee.update(BOB, identity.id, UPDATE_REQUEST)).block())
+    assertThatCode(() => SMono.fromPublisher(testee.update(BOB, customIdentity.id, UPDATE_REQUEST)).block())
       .doesNotThrowAnyException()
   }
 
+  @Test
+  def updateShouldModifiedEntry(): Unit = {
+    when(identityFactory.listIdentities(BOB)).thenReturn(List())
+    val customIdentity: Identity = SMono.fromPublisher(customIdentityDAO.save(BOB, CREATION_REQUEST)).block()
+
+    assertThatCode(() => SMono.fromPublisher(testee.update(BOB, customIdentity.id, UPDATE_REQUEST)).block())
+      .doesNotThrowAnyException()
+
+    assertThat(SFlux(testee.list(BOB)).collectSeq().block().asJava)
+      .containsExactlyInAnyOrder(Identity(id = customIdentity.id,
+        name = IdentityName("Bob (new name)"),
+        email = BOB.asMailAddress(),
+        replyTo = Some(List(EmailAddress(Some(EmailerName("My Boss (updated)")), new MailAddress("boss-updated@domain.tld")))),
+        bcc = Some(List(EmailAddress(Some(EmailerName("My Boss 2 (updated)")), new MailAddress("boss-updated-2@domain.tld")))),
+        textSignature = TextSignature("text 2 signature"),
+        htmlSignature = HtmlSignature("html 2 signature"),
+        mayDelete = MayDeleteIdentity(true)))
+  }
+
+  @Test
+  def updateShouldSuccessWhenMultiUpdateServerSetId(): Unit = {
+    when(identityFactory.listIdentities(BOB)).thenReturn(List(IDENTITY1))
+    SMono.fromPublisher(testee.update(BOB, IDENTITY1.id, UPDATE_REQUEST)).block()
+
+    assertThatCode(() => SMono.fromPublisher(testee.update(BOB, IDENTITY1.id, UPDATE_REQUEST.copy(name = Some(IdentityNameUpdate(IdentityName("Bob (3)")))))).block())
+      .doesNotThrowAnyException()
+
+    assertThat(SFlux(testee.list(BOB)).collectSeq().block().asJava)
+      .containsExactlyInAnyOrder(Identity(id = IDENTITY1.id,
+        name = IdentityName("Bob (3)"),
+        email = BOB.asMailAddress(),
+        replyTo = Some(List(EmailAddress(Some(EmailerName("My Boss (updated)")), new MailAddress("boss-updated@domain.tld")))),
+        bcc = Some(List(EmailAddress(Some(EmailerName("My Boss 2 (updated)")), new MailAddress("boss-updated-2@domain.tld")))),
+        textSignature = TextSignature("text 2 signature"),
+        htmlSignature = HtmlSignature("html 2 signature"),
+        mayDelete = MayDeleteIdentity(true)))
+  }
```

Comment:
Ok now one more test with the second update request updating just one field (like the identity name) and leaves all the other fields untouched.

(If you update all the fields getting the double update right is trivial but partial updates are trickier...)

Reply:
Maybe this is what Benoit suggests?
```scala
@Test
  def updateShouldSuccessWhenSecondPartialUpdateServerSetId(): Unit = {
    when(identityFactory.listIdentities(BOB)).thenReturn(List(IDENTITY1))
    SMono.fromPublisher(testee.update(BOB, IDENTITY1.id, UPDATE_REQUEST)).block()
    val secondUpdateWithName = IdentityUpdateRequest(name = Some(IdentityNameUpdate(IdentityName("Bob (3)"))), None, None,None, None)

    assertThatCode(() => SMono.fromPublisher(testee.update(BOB, IDENTITY1.id, secondUpdateWithName)).block())
      .doesNotThrowAnyException()

    assertThat(SFlux(testee.list(BOB)).collectSeq().block().asJava)
      .containsExactlyInAnyOrder(Identity(id = IDENTITY1.id,
        name = IdentityName("Bob (3)"),
        email = BOB.asMailAddress(),
        replyTo = Some(List(EmailAddress(Some(EmailerName("My Boss (updated)")), new MailAddress("boss-updated@domain.tld")))),
        bcc = Some(List(EmailAddress(Some(EmailerName("My Boss 2 (updated)")), new MailAddress("boss-updated-2@domain.tld")))),
        textSignature = TextSignature("text 2 signature"),
        htmlSignature = HtmlSignature("html 2 signature"),
        mayDelete = MayDeleteIdentity(true)))
  }
```

Reply:
Yes that's it.

Except that we shall not require the `, None, None,None, None` : this boiler plate should be removed by the use of default values...